blueprints/resource/files/app/models/__name__.js

```diff
@@ -1,14 +1,15 @@
+import Ember from 'ember';
 import Resource from 'ember-jsonapi-resources/models/resource';
-import { attr, hasOne, hasMany, hasRelated } from 'ember-jsonapi-resources/models/resource';
+import { attr, hasOne, hasMany } from 'ember-jsonapi-resources/models/resource';
 
 export default Resource.extend({
-  type: '<%= entity %>'
+  type: '<%= entity %>',
+  service: Ember.inject.service('<%= resource %>'),
```

Comment:
I think the issue was that the model wasn't taken from a call to the container, instead it was imported. So adding the `Ember.inject.service` does the trick